controllers/k8ssandracluster_controller.go

```diff
@@ -118,12 +123,16 @@ func (r *K8ssandraClusterReconciler) reconcile(ctx context.Context, kc *api.K8ss
 
 	systemReplication := cassandra.ComputeSystemReplication(kc)
 
-	for i, dcTemplate := range kc.Spec.Cassandra.Datacenters {
+	for _, dcTemplate := range kc.Spec.Cassandra.Datacenters {
 		// Note that it is necessary to use a copy of the CassandraClusterTemplate because
 		// its fields are pointers, and without the copy we could end of with shared
 		// references that would lead to unexpected and incorrect values.
 		dcConfig := cassandra.Coalesce(kc.Spec.Cassandra.DeepCopy(), dcTemplate.DeepCopy())
 		cassandra.ApplySystemReplication(dcConfig, systemReplication)
+		if !strings.HasPrefix(dcConfig.ServerVersion, "3") && kc.HasStargates() {
```

Comment:
There is a `isCassandra4(string)` function in the `cassandra` package FYI. That said, it currently returns true only for version = 4, while your test returns true for version >= 4. Maybe add a `isCassandra3(string)` function and use it in a negated form here?

Reply:
kk, doing this as well